webapp/components/CategoriesSelect/CategoriesSelect.spec.js

```diff
@@ -60,8 +67,8 @@ describe('CategoriesSelect.vue', () => {
         expect(wrapper.vm.selectedCategoryIds).toEqual([categories[0].id])
       })
 
-      it('emits an updateCategories event when the selectedCategoryIds changes', () => {
-        expect(wrapper.emitted().updateCategories[0][0]).toEqual([categories[0].id])
+      it('calls $parent.update with selected category ids', () => {
+        expect(provide.$parentForm.update).toHaveBeenCalledWith('categoryIds', ['cat9'])
```

Comment:
so we should refactor all our code like this?

Reply:
Only custom form elements for `<ds-form>`